File: challenge-load-test-chat/test-by-locust/src/tasks/post_message.py
from locust import task, TaskSet
from endpoints import Endpoints

class PostMessageTask(TaskSet):
    @task
    def post_message(self):
        chat_id = getattr(self.user, 'chat_id', None)
        if chat_id:
            self.client.post(Endpoints.MESSAGE_CREATE.format(chat_id=chat_id), json={"sender": "locust", "content": "hello"})

File: challenge-load-test-chat/test-by-k6/src/tasks/post-message.ts
import http from 'k6/http';
import { check } from 'k6';
import { Endpoints } from '../endpoints';

export function postMessage(baseUrl: string, chatId: string) {
  const payload = JSON.stringify({ sender: 'k6-user', content: 'Hello from k6' });
  const params = { headers: { 'Content-Type': 'application/json' } };
  const res = http.post(`${baseUrl}${Endpoints.MESSAGE_CREATE(chatId)}`, payload, params);
  check(res, { 'post message status is 201': (r) => r.status === 201 });
  return res;
}
